src/components/dialogs/AddToSocialGroupDialog.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useCallback, useState } from 'react';
+import { get } from 'lodash-es';
+import { FormattedMessage } from 'react-intl';
+
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+import Select from '@material-ui/core/Select';
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+
+import useSiteSettings from '../../models/site/useSiteSettings';
+import useSocialGroup from '../../models/socialGroups/useSocialGroup';
+import useSocialGroups from '../../models/socialGroups/useSocialGroups';
+import usePatchSocialGroup from '../../models/socialGroups/usePatchSocialGroup';
+import CustomAlert from '../Alert';
+import InputRow from '../fields/edit/InputRow';
+import Button from '../Button';
+import StandardDialog from '../StandardDialog';
+
+export default function AddToSocialGroupDialog({
+  open,
+  onClose,
+  individualGuid,
+}) {
+  const [selectedGroup, setSelectedGroup] = useState('');
+  const [selectedRole, setSelectedRole] = useState('');
+
+  const { data: socialGroups, loading: socialGroupsLoading } =
+    useSocialGroups();
+  const { data: siteSettings, loading: siteSettingsLoading } =
+    useSiteSettings();
+  const {
+    data: selectedGroupData,
+    loading: selectedGroupDataLoading,
+  } = useSocialGroup(selectedGroup);
+  const {
+    mutate: patchSocialGroup,
+    loading: patchLoading,
+    error: patchError,
+    clearError: clearPatchError,
+  } = usePatchSocialGroup();
+
+  const handleClose = useCallback(() => {
+    setSelectedGroup('');
+    setSelectedRole('');
+    if (patchError) clearPatchError();
+    onClose();
+  }, [patchError]);
+
+  const safeSocialGroups = socialGroups || [];
+  const roles = get(
+    siteSettings,
+    ['social_group_roles', 'value'],
+    [],
+  );
+
+  const formComplete = Boolean(selectedGroup && selectedRole);
+
+  return (
+    <StandardDialog
+      open={open}
+      onClose={handleClose}
+      titleId="ADD_TO_SOCIAL_GROUP"
+      maxWidth="xl"
+    >
+      <DialogContent>
+        <InputRow
+          loading={socialGroupsLoading}
+          schema={{ labelId: 'SOCIAL_GROUP' }}
+        >
+          <FormControl>
+            <InputLabel>
+              <FormattedMessage id="SOCIAL_GROUP" />
+            </InputLabel>
+            <Select
+              style={{ width: 200 }}
+              value={selectedGroup}
+              onChange={e => setSelectedGroup(e.target.value)}
+            >
+              {safeSocialGroups.map(group => (
+                <MenuItem key={group?.guid} value={group?.guid}>
+                  {group?.name}
+                </MenuItem>
```

Comment:
we dont need to prematurely optimize here, the problem shouldn't arise (and if it does, it's okay if the UI is janky to indicate that something problematic is happening). the reason for the `?.`s is to prevent the page from entirely crashing if something unexpected happens